lib/spack/spack/cmd/find.py

```diff
@@ -293,23 +297,27 @@ def root_decorator(spec, string):
         )
         print()
 
-    if args.show_concretized:
+    if args.show_concretized and not args.constraint:
         tty.msg("Concretized roots")
         cmd.display_specs(env.specs_by_hash.values(), args, decorator=decorator)
         print()
```

Comment:
I think we should just remove the concretized roots section entirely, and just make `-c` show all concrete specs in the main output.  I agree with @haampie that it makes the output weird and it's kind of a one-off section.  The roots are shown in the main output and are already highlighted, so we basically show each concretized root twice.

If you can't see the highlighting (it's subtle in some terminals) we could consider underlining them instead of bolding them if folks think that is better.